build_web_compilers/lib/builders.dart

```diff
@@ -194,20 +247,48 @@ bool _readWebHotReloadOption(BuilderOptions options) {
   return options.config[_webHotReloadOption] as bool? ?? false;
 }
 
+List<String> _readAdditionalCoreLibrariesOption(BuilderOptions options) {
+  return (options.config[_additionalCoreLibrariesOption] as List<Object?>?)
+          ?.cast<String>() ??
+      [];
+}
+
+String _readDdcKernelPathOption(BuilderOptions options) {
```

Comment:
We likely want to extend checks in `_ensureSameDdcOptions` to these (if they're in just DDC + kernel builders).

Reply:
_ensureSameDdcOptions` already checks the whole options map so any option is always included in that check.

Reply:
Oh true! And in hindsight, I don't think it makes sense to couple those options. So your current setup is fine.